source/browser.ts

```diff
@@ -371,34 +371,65 @@ ipc.answerMain('reload', () => {
 });
 
 function setDarkMode(): void {
+	setDarkModeElement(document.documentElement);
+	updateVibrancy();
+}
+
+function setDarkModeElement(element: HTMLElement): void {
 	if (is.macos && config.get('followSystemAppearance')) {
 		api.nativeTheme.themeSource = 'system';
 	} else {
 		api.nativeTheme.themeSource = config.get('darkMode') ? 'dark' : 'light';
 	}
 
-	document.documentElement.classList.toggle('dark-mode', api.nativeTheme.shouldUseDarkColors);
-	document.documentElement.classList.toggle('__fb-dark-mode', api.nativeTheme.shouldUseDarkColors);
-
-	updateVibrancy();
+	element.classList.toggle('dark-mode', api.nativeTheme.shouldUseDarkColors);
+	element.classList.toggle('light-mode', !api.nativeTheme.shouldUseDarkColors);
+	element.classList.toggle('__fb-dark-mode', api.nativeTheme.shouldUseDarkColors);
+	element.classList.toggle('__fb-light-mode', !api.nativeTheme.shouldUseDarkColors);
 }
 
 async function observeDarkMode(): Promise<void> {
+	/* Main document's class list */
 	const observer = new MutationObserver((records: MutationRecord[]) => {
 		// Find records that had class attribute changed
 		const classRecords = records.filter(record => record.type === 'attributes' && record.attributeName === 'class');
 		// Check if dark mode classes exists
-		const isDark = classRecords.map(record => {
+		const isDark = classRecords.some(record => {
 			const {classList} = (record.target as HTMLElement);
 			return classList.contains('dark-mode') && classList.contains('__fb-dark-mode');
-		}).includes(true);
+		});
 		// If config and class list don't match, update class list
 		if (api.nativeTheme.shouldUseDarkColors !== isDark) {
 			setDarkMode();
 		}
 	});
 
 	observer.observe(document.documentElement, {attributes: true, attributeFilter: ['class']});
+
+	/* Added nodes (dialogs, etc.) */
+	const observerNew = new MutationObserver((records: MutationRecord[]) => {
+		const nodeRecords = records.filter(record => record.addedNodes.length > 0);
+		for (const nodeRecord of nodeRecords) {
+			for (const newNode of nodeRecord.addedNodes) {
+				const {classList} = (newNode as HTMLElement);
+				const isLight = classList.contains('light-mode') || classList.contains('__fb-light-mode');
+				if (api.nativeTheme.shouldUseDarkColors === isLight) {
+					setDarkModeElement(newNode as HTMLElement);
+				}
+			}
+		}
+	});
+
+	/* Observe only elements where new nodes may need dark mode */
+	const menuElements = await elementReady<HTMLElement>('div[class="rq0escxv l9j0dhe7 du4w35lb"] > div:nth-of-type(3) > div', {stopOnDomReady: false});
```

Comment:
The query selectors are inverted and I'd prefer using dot notation for classes instead of attribute notation. That way if in some case they are rendered with different order it would still work but with attribute notation it won't.

```suggestion
	const modalElements = await elementReady<HTMLElement>('div.rq0escxv.l9j0dhe7.du4w35lb > div:nth-of-type(3) > div', {stopOnDomReady: false});
```

Reply:
The issue with the dot notation here is that there are many partial matches to `.rq0escxv.l9j0dhe7.du4w35lb`. Even with an exact match, there are still multiple elements that match. If we rely on the child structure `> div:nth-of-type(3) > div`, then there is only one match, regardless of partial/complete match. However, I was wanting to do an exact match, just in case the structure of partial matches changes later.

With the preferences overlay open:
```javascript
> document.querySelectorAll('.rq0escxv.l9j0dhe7.du4w35lb').length
<- 151
> document.querySelectorAll('div.rq0escxv.l9j0dhe7.du4w35lb').length
<- 132
> document.querySelectorAll('div[class="rq0escxv l9j0dhe7 du4w35lb"]').length
<- 3
```

Going with `'div.rq0escxv.l9j0dhe7.du4w35lb > div:nth-of-type(3) > div'` works for now, but is there a way to avoid partial matches with the dot notation?

Reply:
Yeah, you're right. I don't know any good way to use dot notation and exclude partial matches so just leave it with attribute notation and change the variable name 😊.